src/vision/shiny_future.md

```diff
@@ -0,0 +1,66 @@
+
+## Sally builds a web server
+
+Sally has decided to build a web application. She's familiar with async-await from JavaScript, and she knows that Rust has a similar feature. She's eager to get started, so she goes to "Rust by Example" where she sees an example for building a web server that runs the guessing game. That sounds perfect!
+
+Following the book's instructions, she quickly stands up some simple examples using async-await. She's using the standard Rust runtime and recommended web framework. With only a few line of code, she's able to play the ["guessing game"] in her browser[^written-in-rust]. This is fun!
+
+["guessing game"]: https://doc.rust-lang.org/book/ch02-00-guessing-game-tutorial.html
+
+**Key observations:**
+
+* Async is integrated into standard Rust learning materials like RBE, not its own silo.
+* Standard ways to stand up a runtime and web server, no choices to make (at first...)
+    * *FIXME:* We should cover how she is able to easily change to other runtimes later and things keep working
+
+## Sally learns about spawn
+
+Still playing around, Sally starts to extend her 'guessing game' project with some new functionality. She adds a function that launches a few web requests to help her pick the random number and get a nice "quote of the day" to include in the website:
+
+```rust,ignore
+async fn guessing_game_configuration() -> std::io::Result<(usize, String)> {
+    let random_number = async_io::request::fetch("https://random-number.org/").await?;
+    let quotd = async_io::request::fetch("https://qotd.org/").await?;
+    Ok((random_number, qotd))
+}
+```
+
+Here, she is using the standard API for launching web requests. When she puts this into her IDE, she gets a lint warning:
+
+```notrust
+warning[E2222]: synchronized request fetch
+ --> src/main.rs:2:13
+  |
+2 |     let random_number = async_io::request::fetch("https://random-number.org/").await?;
+  |                                                                                ------
+3 |     let quotd = async_io::request::fetch("https://qotd.org/").await?;
+  |                                                               ------
+  |
+  = help: these two request fetches will be synchronized, which probably isn't what you want
+2 |     let random_number = async_io::request::fetch("https://random-number.org/").spawn().await?;
+  |                                                                                ----- consider introducing calls to `spawn`
+3 |     let quotd = async_io::spawn(async_io::request::fetch("https://qotd.org/")).spawn().await?;
+  |                                                                                ----- here too
+```
+
+Reading a bit more into the background for E2222, she learns about how async-await in Rust works differently from async-await in javaScript. She learns about spawning. etc etc.
```

Comment:
The issue of serial `await`s would have had the same problem in JavaScript. In JS, the user would have needed to use `Promise.all` to await them concurrently. In this proposed adjustment, even with spawning, awaiting the spawned task would probably _still_ result in serial tasks (the second spawn wouldn't get triggered until the first spawned task completed).